### PR TITLE
OPS: upgrade to gradle 7.5.1

### PR DESCRIPTION
**Pull request by joe-chacko**

### CODE REVIEW

**Review by ngmr — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 1]
```diff
@@ -48,33 +54,28 @@ subprojects {
...
 
-  configurations {
-    // declare a new configuration to map to Maven's provided scope
-    provided.transitive = true
-    compile.extendsFrom provided
-    // declare a new configuration to use for spec API jars
-    spec.transitive = true
-    compile.extendsFrom spec
-    // declare a test library configuration
-    testLib
```
[the remark this answers is not in the document]
Comment (joe-chacko) [reply build.gradle:59]: done

File: common.bnd [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Bundle-DocURL: https://geronimo.apache.org/maven/yoko/${project.version}/${project.name}
```
[the remark this answers is not in the document]
Comment (joe-chacko) [reply common.bnd:15]: done

File: yoko-core/build.gradle [hunk 3]
```diff
@@ -1,44 +1,42 @@
...
-  test.java.srcDir "src/test/java-testify"  // new framework tests, segregated during transition
+  test.java.srcDir "src/test/java-testify"  // separate ported tests until porting is complete
   test.compileClasspath += artifact.output
 }
 
 testClasses.dependsOn v0Classes, v1Classes, v2Classes
 
 configurations {
-  artifactImplementation.extendsFrom compile, testLib
+  artifactImplementation.extendsFrom implementation, testLib
```
[the remark this answers is not in the document]
Comment (joe-chacko) [reply yoko-core/build.gradle:14]: correct

File: yoko-core/build.gradle [hunk 4]
```diff
@@ -1,44 +1,42 @@
...
   v0Implementation.extendsFrom artifactImplementation
   v1Implementation.extendsFrom artifactImplementation
   v2Implementation.extendsFrom artifactImplementation
+  testImplementation.extendsFrom testLib
```
[the remark this answers is not in the document]
Comment (joe-chacko) [reply yoko-core/build.gradle:18]: correct

File: yoko-core/build.gradle [hunk 5]
```diff
@@ -1,44 +1,42 @@
...
-  // compile against Yoko implementation
-  compile project(":yoko-rmi-impl")
+  // and the rest of the Yoko implementation
+  implementation project(":yoko-util")
+  implementation project(":yoko-rmi-impl")
 
   artifactImplementation sourceSets.main.output
 
   // use shared test utilities
+  testLib "org.hamcrest:hamcrest:2.1"
```
[the remark this answers is not in the document]
Comment (joe-chacko) [reply yoko-core/build.gradle]: correct

File: yoko-core/build.gradle [hunk 6]
```diff
@@ -50,6 +48,7 @@ test {
 task debug() {
   doLast {
     sourceSets.each { srcSet ->
+      if ("test" != srcSet.name) return
```
[the remark this answers is not in the document]
Comment (joe-chacko) [reply yoko-core/build.gradle]: done

File: yoko-test/build.gradle [hunk 7]
```diff
@@ -5,42 +5,20 @@ repositories {
...
-       */
-      'Bundle-Description': 'TestOSGi',
-      'Bundle-Name': project.name,
-      // replace "yoko-" prefix in project name with full group name
-      'Bundle-SymbolicName': project.name.replaceFirst("^yoko-", project.group.toString() + "."),
-      'Import-Package': '*',
-      'Test-Cases': '${classes;CONCRETE;PUBLIC;NAMED;*Test}'
-    )
-  }
+  implementation project(":yoko-spec-corba"), project(':yoko-core')
```
[the remark this answers is not in the document]
Comment (joe-chacko) [reply yoko-test/build.gradle:12]: agree

[the remark this answers is not in the document]
Comment (joe-chacko) [reply yoko-test/build.gradle:12]: done

File: yoko-spec-corba/bnd.bnd [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+-include ~../common.bnd
+
+Bundle-Description: Apache Yoko CORBA Spec
+Bundle-Activator: org.apache.yoko.Activator
+Export-Package: \
+  org.omg.*;version=2.4, \
+  org.apache.geronimo.interop;version=1.0, \
+  org.apache.yoko;version=1.0
+
+-buildpath: \
```
[the remark this answers is not in the document]
Comment (joe-chacko) [reply yoko-spec-corba/bnd.bnd]: done

File: build.gradle [hunk 9]
```diff
@@ -200,12 +188,6 @@ configure(shipProjects) {
       from 'build/publications/maven'
       rename "pom-default.xml", "pom.xml"
     }
-    manifest {
-      instruction 'Bundle-DocURL'          , "https://geronimo.apache.org/maven/yoko/$project.version/$project.name"
-      instruction 'Bundle-License'         , 'https://www.apache.org/licenses/LICENSE-2.0.txt'
-      instruction 'Bundle-Vendor'          , 'The Apache Software Foundation'
-      instruction 'Import-Package'         , 'sun.*;resolution:=optional,*'
```
[the remark this answers is not in the document]
Comment (joe-chacko) [reply build.gradle:207]: Proved: 
```
bnd **/libs/*.jar | grep 'sun\.'
                                        sun.misc;resolution:=optional
                                        sun.reflect;resolution:=optional
```

This is handled inside rmi-impl's build.gradle, which is where these packages are used.

File: build.gradle [hunk 10]
```diff
@@ -168,20 +166,10 @@ configure(shipProjects) {
           def depsNode = rootNode.dependencies[0] ?:
                   rootNode.appendNode('dependencies')
 
-          // declare 'provided' scope dependencies in the pom.xml
```
[the remark this answers is not in the document]
Comment (joe-chacko) [reply build.gradle:171]: The dependencies are correct in the new versions, but some duplication has been lost.
The 'provided' scope is missing, but I don't think this matters.

**Review by ngmr — APPROVED** [review]
As addressing the commented out 'gradle.buildFinished' stanza will be addressed (removed) in a subsequent issue, I'm happy with the change.

**Pull request merged**